File: 4_Insertion_Sort/InsertionSort.java
//Insertion sort algorithm implementation
public class InsertionSort {
    public static int[] insertionSort(int[] arr) {
        int n = arr.length;
        for (int i = 1; i < n; i++) {
            int key = arr[i]; // element at 1 postion for 1st time run
            int j = i - 1;
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }
        return arr;
    }

    // driver code
    public static void main(String[] args) {
        int[] array = { 12, 11, 13, 5, 6 };
        int[] result = insertionSort(array);
        for (int element : result) {
            System.out.printf(element + " ");
        }
    }
}

File: 4_Insertion_Sort/InsertionSort.py
def insertionSort(arr):
    n = len(arr)
    for i in range(1, n):
        key = arr[i]
        j = i-1
        while(j >= 0 and arr[j] > key):
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key

# Driver code to test above
arr = [8, 9, 6, 10, 1, 2]

insertionSort(arr)
  
print("Sorted array is:")
for i in range(len(arr)):
    print("%d" % arr[i], end=" ")